components/family-studies/migrations/src/main/java/org/phenotips/studies/family/migrations/R71500PhenoTips2944DataMigration.java

```diff
@@ -0,0 +1,311 @@
+                    continue;
+                }
+
+                BaseObject pedigreeXObject = familyXDocument.getXObject(PEDIGREE_CLASS_REFERENCE);
+                if (pedigreeXObject == null) {
+                    continue;
+                }
+
+                this.logger.debug("Updating pedigree for family {}.", docName);
+                updated = this.updatePedigreeGenes(pedigreeXObject, context, docName);
+                if (!updated) {
+                    continue;
+                }
+
+                familyXDocument.setComment(this.getDescription());
+                familyXDocument.setMinorEdit(true);
+
+            } catch (Exception e) {
+                this.logger.error("Error converting gene data for family {}: [{}]", docName, e.getMessage());
+                continue;
+            }
+
+            try {
+                // There's a bug in XWiki which prevents saving an object in the same session that it was loaded,
+                // so we must clear the session cache first.
+                session.clear();
+                ((XWikiHibernateStore) getStore()).saveXWikiDoc(familyXDocument, context, false);
+                session.flush();
+            } catch (DataMigrationException e) {
+                //
+            }
+
+        }
+        return null;
+    }
+
+    private boolean updatePedigreeGenes(BaseObject pedigreeXObject, XWikiContext context, String documentName)
+    {
+        boolean updatedGenes = false;
+        boolean updatedColors = false;
+        String dataText = pedigreeXObject.getStringValue(PEDIGREECLASS_JSONDATA_KEY);
+        if (StringUtils.isEmpty(dataText)) {
+            return false;
+        }
+
+        try {
+            JSONObject pedigree = new JSONObject(dataText);
+            Map<String, String> hgncToEnsemblMap = new HashMap<String, String>();
+
+            JSONArray pedigreeNodes = pedigree.optJSONArray(PEDIGREE_GRAPH_KEY);
+            if (pedigreeNodes != null) {
+                for (Object node : pedigreeNodes) {
+                    JSONObject nodeJSON = (JSONObject) node;
+                    updatedGenes = this.convertGenes(nodeJSON, hgncToEnsemblMap);
+                }
+            }
+
+            JSONObject pedigreeSettings = pedigree.optJSONObject(PEDIGREE_SETTINGS_KEY);
+            if (pedigreeSettings != null) {
+                JSONObject pedigreeColors = pedigreeSettings.optJSONObject(PEDIGREE_COLORS_FIELD);
+                if (pedigreeColors != null) {
+                    updatedColors = this.convertColors(pedigreeColors, hgncToEnsemblMap);
+                }
+            }
+
+            if (updatedGenes || updatedColors) {
+                String pedigreeData = pedigree.toString();
+                pedigreeXObject.set(PEDIGREECLASS_JSONDATA_KEY, pedigreeData, context);
+            }
+        } catch (Exception e) {
+            this.logger.error("Family pedigree data is not a valid JSON for family {}: [{}]", documentName, e);
```

Comment:
we don't know the exact problem, maybe conversion fails for some other reason (e.g. vocabulary throws an exception) so I would just say "error converting"